core/types/transaction_signing.go

```diff
@@ -347,3 +347,17 @@ func DeriveChainId(v *uint256.Int) *uint256.Int {
 	v = new(uint256.Int).Sub(v, u256.Num35)
 	return v.Div(v, u256.Num2)
 }
+
+// Hash returns the hash to be signed by the sender.
+// It does not uniquely identify the transaction.
+func (sg Signer) Hash(tx Transaction) common.Hash {
```

Comment:
can you take a look at `SigningHash` function of type `LegacyTx` in the file `core/legacy_tx.go`. If you only need this function for legacy-type txs, you can try to use existing function?